mongodb-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/MongoDBBatchSink.java

```diff
@@ -57,7 +59,13 @@ public MongoDBBatchSink(MongoDBSinkConfig config) {
 
   @Override
   public void prepareRun(BatchSinkContext context) throws Exception {
-    context.addOutput(Output.of(config.referenceName, new MongoDBOutputFormatProvider(config)));
+    Configuration conf = new Configuration();
+    String path = conf.get(
+      "mapreduce.task.tmp.dir",
+      conf.get(
+        "mapred.child.tmp",
+        conf.get("hadoop.tmp.dir", "/tmp"))) + "/" + UUID.randomUUID().toString();
```

Comment:
Should never use `/tmp` hardcoded. Use `System.getProperty("java.io.tmpdir")` instead. It will be set to local container directory.